extensions/amp-next-page/1.0/service.js

```diff
@@ -471,12 +630,49 @@ export class NextPageService {
   }
 
   /**
+   * Parses the amp-next-page element for inline or remote list of pages and
+   * add them to the queue
+   * @param {!Element=} element the container of the amp-next-page extension
+   * @private
+   */
+  parseAndQueuePages_(element = this.getHostNextPageElement_()) {
+    this.parsePages_(element).then(pages => {
+      pages.forEach(page => {
+        try {
+          validatePage(page, this.ampdoc_.getUrl());
+          // Prevent loops by checking if the page already exists
+          // we use initialUrl since the url can get updated if
+          // the page issues a redirect
+          if (this.pages_.some(p => p.initialUrl == page.url)) {
+            user().warn(TAG, 'Recommendation already queued');
```

Comment:
Is this warning necessary? I would say it's OK to ignore duplicates silently since that is a stated behavior of the component.